components/Frame/ProlongBox.js

```diff
@@ -25,28 +29,67 @@ const styles = {
     backgroundColor: colors.primaryBg
   }),
   boxDark: css({
-    backgroundColor: negativeColors.primaryBg
+    backgroundColor: negativeColors.primaryBg,
+    color: '#fff'
   })
 }
 
+const SingleLine = ({ children }) => <div {...styles.singleLine}>
+  {children}
+</div>
+
+const dayFormat = timeFormat('%d. %B %Y')
+
 const ProlongBox = ({
   t, prolongBeforeDate, router,
-  inNativeApp, inNativeIOSApp, dark
+  inNativeApp, inNativeIOSApp, dark: inDarkFrame
 }) => {
   if (router.pathname === '/pledge' || router.pathname === '/cancel' || router.pathname === '/meta') {
     return null
   }
   const date = new Date(prolongBeforeDate)
   const numberOfDays = timeDay.count(new Date(), date)
   if (numberOfDays <= 30) {
+    const key = numberOfDays <= 2
+      ? numberOfDays < 0
+        ? 'overdue'
+        : 'due'
+      : 'before'
+    const baseKey = `prolongNecessary/${key}`
+
+    const dark = inDarkFrame || key !== 'before'
+    const colorStyle = { color: dark ? '#fff' : undefined }
+
+    const explanation = t.elements(`${baseKey}/explanation`, {
+      cancelLink: <Link key='cancelLink' route='cancel' passHref>
+        <Editorial.A style={colorStyle}>
+          {t(`${baseKey}/explanation/cancelText`)}
+        </Editorial.A>
+      </Link>,
+      graceEndDate: dayFormat(timeDay.offset(date, 14))
```

Comment:
Shall we deliver that `graceEndDate` that via API to ensure backend and frontend are in sync?